web/src/app.tsx

```diff
@@ -33,7 +33,8 @@ export async function getInitialState(): Promise<{
 }> {
   const token = localStorage.getItem('token');
   if (!token) {
-    history.replace(`/user/login?redirect=${encodeURIComponent(window.location.pathname)}`);
+    const redirect = getUrlQuery('redirect');
```

Comment:
![image](https://user-images.githubusercontent.com/2561857/128652574-9c28fdc2-a1c8-4c1d-b7b8-f26210f93e50.png)

hi @foolwc , when the url did not have redirect param, we should omit it, and redirect to `/`

Reply:
Got it. Thanks a lot